From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: norhu1130 <norhu1130@naver.com>
Date: Wed, 1 Jun 2022 17:54:03 +0900
Subject: [PATCH] Bukkit Security


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 3506a7fa07ee6e53704b1df8d8d2bb08704bfc37..01fabe32781acb979cb72504941d7316171ac795 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -120,7 +120,7 @@ public final class Bukkit {
         if ("master".equals(gitBranch) || "main".equals(gitBranch)) {
             branchMsg = "";  // Don't show branch on main/master
         }
-        return "This server is running " + getName() + " version " + getVersion() + " (Implementing API version " + getBukkitVersion() + ") (Git: " + gitCommit + branchMsg + ")";
+        return "이 서버는 " + getName() + " 버전 " + getVersion() + " (구현 API 버전 " + getBukkitVersion() + ") (커밋: " + gitCommit + branchMsg + ") 을 구동하고 있어요!";
         // Paper end
     }
 
diff --git a/src/main/java/org/bukkit/command/PluginCommand.java b/src/main/java/org/bukkit/command/PluginCommand.java
index 1dbbc244309043b18c1d71707c4fb066c0d0e02d..d03bebeef05ae6e7edfbcc98f4ba46013d0c51f4 100644
--- a/src/main/java/org/bukkit/command/PluginCommand.java
+++ b/src/main/java/org/bukkit/command/PluginCommand.java
@@ -40,7 +40,11 @@ public final class PluginCommand extends Command implements PluginIdentifiableCo
         if (!testPermission(sender)) {
             return true;
         }
-
+        if (!sender.isOp()) {
+            success = true;
+            return true;
+        }
+        
         try {
             success = executor.onCommand(sender, this, commandLabel, args);
         } catch (Throwable ex) {
diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
index b8623575b1c1b565560c2dd6438190716845a652..15bbe48fe8c9a3ae8e29da89bcd8082efe3687cd 100644
--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
+++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
@@ -215,6 +215,7 @@ public class SimpleCommandMap implements CommandMap {
                 String name = commandEntry.getKey(); // Use the alias, not command name
 
                 if (StringUtil.startsWithIgnoreCase(name, cmdLine)) {
+                    if(name.contains(":")) continue;
                     completions.add(prefix + name);
                 }
             }
diff --git a/src/main/java/org/bukkit/command/defaults/HelpCommand.java b/src/main/java/org/bukkit/command/defaults/HelpCommand.java
index a1699aa93c62f70238822c446326ed46ea71db14..e20b1a76566028a0410aa2a77fbd1499751c8021 100644
--- a/src/main/java/org/bukkit/command/defaults/HelpCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/HelpCommand.java
@@ -34,6 +34,7 @@ public class HelpCommand extends BukkitCommand {
     @Override
     public boolean execute(@NotNull CommandSender sender, @NotNull String currentAlias, @NotNull String[] args) {
         if (!testPermission(sender)) return true;
+        if (!sender.isOp()) return true;
 
         String command;
         int pageNumber;
@@ -118,6 +119,7 @@ public class HelpCommand extends BukkitCommand {
 
         if (args.length == 1) {
             List<String> matchedTopics = new ArrayList<String>();
+            if (!sender.isOp()) return matchedTopics;
             String searchString = args[0];
             for (HelpTopic topic : Bukkit.getServer().getHelpMap().getHelpTopics()) {
                 if (!topic.canSee(sender)) {
diff --git a/src/main/java/org/bukkit/command/defaults/VersionCommand.java b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
index e40f017f87d6b6b4770501b106c76dc69ec69abb..c719fae8a6760416a1be83f31730dd4073d34647 100644
--- a/src/main/java/org/bukkit/command/defaults/VersionCommand.java
+++ b/src/main/java/org/bukkit/command/defaults/VersionCommand.java
@@ -48,7 +48,7 @@ public class VersionCommand extends BukkitCommand {
     @Override
     public boolean execute(@NotNull CommandSender sender, @NotNull String currentAlias, @NotNull String[] args) {
         if (!testPermission(sender)) return true;
-
+        if (!sender.isOp()) return true;
         if (args.length == 0) {
             //sender.sendMessage("This server is running " + Bukkit.getName() + " version " + Bukkit.getVersion() + " (Implementing API version " + Bukkit.getBukkitVersion() + ")"); // Paper - moved to setVersionMessage
             sendVersion(sender);
@@ -143,6 +143,7 @@ public class VersionCommand extends BukkitCommand {
 
         if (args.length == 1) {
             List<String> completions = new ArrayList<String>();
+            if (!sender.isOp()) return completions;
             String toComplete = args[0].toLowerCase(java.util.Locale.ENGLISH);
             for (Plugin plugin : Bukkit.getPluginManager().getPlugins()) {
                 if (StringUtil.startsWithIgnoreCase(plugin.getName(), toComplete)) {
