From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: norhu1130 <norhu1130@naver.com>
Date: Tue, 31 May 2022 22:30:00 +0900
Subject: [PATCH] Fix Tiny Remapper


diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java b/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
index 26345494ce190b5cd2ab58dd7d4b046796767b20..ab44d43a4a5698914c209020b4d39b39f87a223c 100644
--- a/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
+++ b/src/main/java/net/minecraft/server/dedicated/DedicatedServerProperties.java
@@ -155,7 +155,7 @@ public class DedicatedServerProperties extends Settings<DedicatedServerPropertie
         }, WorldPresets.NORMAL.location().toString()));
         this.serverResourcePackInfo = DedicatedServerProperties.getServerPackInfo(this.get("resource-pack", ""), this.get("resource-pack-sha1", ""), this.getLegacyString("resource-pack-hash"), this.get("require-resource-pack", false), this.get("resource-pack-prompt", ""));
         // Paper start - Configurable rcon ip
-        final String rconIp = this.getStringRaw("rcon.ip");
+        final String rconIp = this.getSettingIfExists("rcon.ip");
         this.rconIp = rconIp == null ? this.serverIp : rconIp;
         // Paper end
     }
diff --git a/src/main/java/net/minecraft/server/dedicated/Settings.java b/src/main/java/net/minecraft/server/dedicated/Settings.java
index aafa84578c7fb25feeee043259f9c056929ca008..79c65629890c9ebafca55b7022d37bb7853fdd14 100644
--- a/src/main/java/net/minecraft/server/dedicated/Settings.java
+++ b/src/main/java/net/minecraft/server/dedicated/Settings.java
@@ -34,7 +34,7 @@ public abstract class Settings<T extends Settings<T>> {
     }
 
     private String getOverride(String name, String value) {
-        if ((this.options != null) && (this.options.has(name)) && !name.equals( "online-mode")) { // Spigot
+        if ((this.options != null) && (this.options.has(name))) { // Spigot
             return String.valueOf(this.options.valueOf(name));
         }
 
@@ -163,13 +163,13 @@ public abstract class Settings<T extends Settings<T>> {
     }
 
     @Nullable
-    public String getStringRaw(String key) {
-        return (String) this.getOverride(key, this.properties.getProperty(key)); // CraftBukkit
+    public String getSettingIfExists(String key) {
+        return this.getOverride(key, this.properties.getProperty(key)); // CraftBukkit
     }
 
     @Nullable
     protected <V> V getLegacy(String key, Function<String, V> stringifier) {
-        String s1 = this.getStringRaw(key);
+        String s1 = this.getSettingIfExists(key);
 
         if (s1 == null) {
             return null;
@@ -180,7 +180,7 @@ public abstract class Settings<T extends Settings<T>> {
     }
 
     protected <V> V get(String key, Function<String, V> parser, Function<V, String> stringifier, V fallback) {
-        String s1 = this.getStringRaw(key);
+        String s1 = this.getSettingIfExists(key);
         V v1 = MoreObjects.firstNonNull(s1 != null ? parser.apply(s1) : null, fallback);
 
         this.properties.put(key, stringifier.apply(v1));
@@ -188,7 +188,7 @@ public abstract class Settings<T extends Settings<T>> {
     }
 
     protected <V> Settings<T>.MutableValue<V> getMutable(String key, Function<String, V> parser, Function<V, String> stringifier, V fallback) {
-        String s1 = this.getStringRaw(key);
+        String s1 = this.getSettingIfExists(key);
         V v1 = MoreObjects.firstNonNull(s1 != null ? parser.apply(s1) : null, fallback);
 
         this.properties.put(key, stringifier.apply(v1));
